### PR TITLE
Replace calls to getProjectAndPackageIds() with collectProjectsAndPackages()

### PR DESCRIPTION
**Pull request by sschuberth**
Please have a look at the individual commit messages for the details.

### CODE REVIEW

**Review by oheger-bosch — DISMISSED** [review]

**Comment by sschuberth** [comment]
@fviernau are you fine with merging this, or do you use `getProjectAndPackageIds()` in rules?

File: model/src/main/kotlin/OrtResult.kt [hunk 1]
```diff
@@ -211,13 +211,14 @@ data class OrtResult(
      */
     fun collectProjectsAndPackages(includeSubProjects: Boolean = true): Set<Identifier> {
         val projectsAndPackages = mutableSetOf<Identifier>()
+        val projects = getProjects()
```
Comment (mnonnenmacher) [inline model/src/main/kotlin/OrtResult.kt:214]: Commit message: "project**s**"

File: model/src/main/kotlin/OrtResult.kt [hunk 2]
```diff
@@ -218,8 +218,8 @@ data class OrtResult(
         if (!includeSubProjects) {
             val allSubProjects = mutableSetOf<Identifier>()
 
-            projects.forEach {
-                allSubProjects += dependencyNavigator.collectSubProjects(it)
+            projects.flatMapTo(allSubProjects) {
```
Comment (mnonnenmacher) [inline model/src/main/kotlin/OrtResult.kt:221]: Could also be collapsed to `val allSubProjects = projects.flatMapTo(mutableSetOf()) {`.

File: model/src/main/kotlin/DependencyNavigator.kt [hunk 3]
```diff
@@ -161,7 +160,7 @@ interface DependencyNavigator {
         project: Project,
         maxDepth: Int = -1,
         matcher: DependencyMatcher = MATCH_ALL
-    ): SortedSet<Identifier> =
+    ): Set<Identifier> =
```
Comment (mnonnenmacher) [inline model/src/main/kotlin/DependencyNavigator.kt:163]: Did you verify that all callers of this and the above function either sort on their own or don't require sorting?

**Review by mnonnenmacher — CHANGES_REQUESTED** [review]

File: model/src/main/kotlin/DependencyNavigator.kt (hunk 3 above)
Comment (sschuberth) [reply model/src/main/kotlin/DependencyNavigator.kt:163]: Yes, and I also consulted @oheger-bosch for this.

**Comment by fviernau** [comment]
> @fviernau are you fine with merging this, or do you use `getProjectAndPackageIds()` in rules?

Can be merged from our internal setup perspective.

File: helper-cli/src/main/kotlin/common/Utils.kt [hunk 4]
```diff
@@ -239,7 +239,7 @@ internal fun OrtResult.processAllCopyrightStatements(
         copyrightGarbage = CopyrightGarbage(copyrightGarbage.toSortedSet())
     )
 
-    getProjectAndPackageIds().forEach { id ->
+    collectProjectsAndPackages().forEach { id ->
```
Comment (fviernau) [inline helper-cli/src/main/kotlin/common/Utils.kt:242]: commit-msg: May indicate whether the change is intended to be non-functional.

Comment (fviernau) [inline helper-cli/src/main/kotlin/common/Utils.kt:242]: commit-msg: Please indicate whether the change is intended to be non-functional.

**Review by fviernau — CHANGES_REQUESTED** [review]

File: helper-cli/src/main/kotlin/common/Utils.kt (hunk 4 above)
Comment (sschuberth) [reply helper-cli/src/main/kotlin/common/Utils.kt:242]: Done.

**Review by fviernau — APPROVED** [review]

**Pull request merged**